src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -187,9 +191,9 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
-            // Randomly order the list of peers and return the amount asked for.
-            var allPeers = this.peerAddresses.OrderBy(p => this.random.Next());
-            return allPeers.Select(p => p.Value).Take(1000);
+            var discoverable = this.peerAddresses.Values.Where(p => p.LastDiscoveredFrom < DateTimeProvider.Default.GetUtcNow().AddHours(-PeerSelector.DiscoveryThresholdHours));
```

Comment:
You should probably use here the injected property `DateTimeProvider.Default`

Reply:
Fixed 👍